clarity_ext/extensions.py

```diff
@@ -258,10 +260,11 @@ def _run(self, config, path, pid, module, artifacts_to_stdout, disable_context_c
                                           disable_commits=disable_context_commit,
                                           uploaded_to_stdout=artifacts_to_stdout)
         instance = extension(context)
-        context.logger.log("Start running script '{}'".format(module))
-        # We add the config in this way (not via __init__) as a quick fix.
-        instance.config = config
         try:
+            context.logger.log("Start running script '{}'".format(module))
+            context.validate_xml_concordance()
```

Comment:
How about just doing the validation when the data is required? I'm thinking e.g. pushing it into the `all_artifacts` method, or similar. That is, as close to the actual call to the genologics library.

I think that might make it easier to reason about and remove afterwards. A minor effect is that it could perform better, in the case we have an extension that doesn't require all artifacts.